### PR TITLE
Use mocha and gulp

### PR DESCRIPTION
**Pull request by neomatrix369**
Applying changes for issue #6 
- applied changes to migrate to mocha
- applied changes to migrate to gulp

### CODE REVIEW

**Comment by neomatrix369** [comment]
@julianghionoiu Lets start discussing this PR - I have run the nodejs docker code coverage tests and here are the results:

```bash
~~~ Summary of test executions ~~~
  ~~~ Passed Tests ~~~
  language=nodejs|challenge=SUM|expected=100|actual=100
  language=nodejs|challenge=TST|expected=75|actual=75
  language=nodejs|challenge=CHK|expected=0|actual=0
  language=nodejs|challenge=xyz|expected=0|actual=0
  4 test(s) passed

  ~~~ Failed Tests ~~~
  0 test(s) failed
```

**Comment by neomatrix369** [comment]
You mentioned the test coverage scripts might need changing, but I didn't see much change was needed.

File: gulpfile.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+var gulp = require('gulp');
+var run = require('gulp-run-command').default;
```
Comment (julianghionoiu) [inline gulpfile.js]: You are right about Gulp, in this case, it is not adding any value.

We should remove it, so just Mocha with NPM

File: package.json [hunk 2]
```diff
@@ -4,24 +4,27 @@
...
+    "gulp": "^4.0.0",
+    "mocha": "^5.2.0",
     "properties-reader": "0.0.15",
     "tdl-client-nodejs": "^0.25.0",
-    "unirest": "0.5.1",
-    "diveSync": "0.3.0"
+    "unirest": "0.5.1"
   },
   "devDependencies": {
     "ava": "^0.25.0",
```
Comment (julianghionoiu) [inline package.json]: We also need to remove the unneeded dependencies

File: test/solutions/TST/test-some-numbers.js [hunk 3]
```diff
@@ -1,8 +1,14 @@
...
 const one = require('../../../lib/solutions/TST/one');
 // noinspection JSUnusedLocalSymbols
 const two = require('../../../lib/solutions/TST/two');
 
-test("show one", function(t){
-    t.is(one(), 1);
-});
+describe('TST challenge', function() {
+	it('show one', function() {
+		assert.equals(one(), 1);
```
Comment (julianghionoiu) [inline test/solutions/TST/test-some-numbers.js]: When I ran `npm test`, I got an error about the assert:
```
➜  tdl-runner-nodejs git:(use-mocha-and-gulp) npm test

> tdl-warmup@0.0.1 test /Users/constantin-iulianghionoiu/Documents/Work/Projects/tdl/tdl-runner-nodejs
> node ./node_modules/mocha/bin/mocha test/**/*.js



  SUM challenge: adding two numbers
    1) should return 3, which is the sum of 1 and 2

  TST challenge
    2) show one


  0 passing (17ms)
  2 failing

  1) SUM challenge: adding two numbers
       should return 3, which is the sum of 1 and 2:
     Error: method not implemented
      at module.exports (lib/solutions/SUM/sum.js:4:11)
      at Context.<anonymous> (test/solutions/SUM/test-sum.js:9:20)

  2) TST challenge
       show one:
     TypeError: assert.equals is not a function
      at Context.<anonymous> (test/solutions/TST/test-some-numbers.js:12:10)

```

**Comment by neomatrix369** [comment]
@julianghionoiu let me know what you think of these changes, test results attache to PR

File: test/solutions/TST/test-some-numbers.js (hunk 3 above)
Comment (neomatrix369) [reply test/solutions/TST/test-some-numbers.js]: Fixed via 2defc20

File: package.json (hunk 2 above)
Comment (neomatrix369) [reply package.json]: Fixed via 898f54a

File: gulpfile.js (hunk 1 above)
Comment (neomatrix369) [reply gulpfile.js]: We never had any gulp changes, can you find them in the PR ;) :P

**Comment by neomatrix369** [comment]
```
~~~ Summary of test executions ~~~
  ~~~ Passed Tests ~~~
  language=nodejs|challenge=SUM|expected=100|actual=100
  language=nodejs|challenge=TST|expected=75|actual=75
  language=nodejs|challenge=CHK|expected=0|actual=0
  language=nodejs|challenge=xyz|expected=0|actual=0
  4 test(s) passed

  ~~~ Failed Tests ~~~
  0 test(s) failed
```

**Comment by neomatrix369** [comment]
```
± npm test

> tdl-warmup@0.0.1 test /path/to/tdl-runner-nodejs
> node ./node_modules/mocha/bin/mocha test/**/*.js



  SUM challenge: adding two numbers
    1) should return 3, which is the sum of 1 and 2

  TST challenge
    ✓ show one


  1 passing (13ms)
  1 failing

  1) SUM challenge: adding two numbers
       should return 3, which is the sum of 1 and 2:
     Error: method not implemented
      at exports (lib/solutions/SUM/sum.js:4:11)
      at Context.<anonymous> (test/solutions/SUM/test-sum.js:9:19)



npm ERR! Test failed.  See above for more details.
```

don't know what you are referring to ;)

**Review by julianghionoiu — APPROVED** [review]

**Pull request merged**